### PR TITLE
Minimal numpy version

### LINKED ISSUE #8: numpy version 1.19.5 not working with pyoscode
Just tried installing pyoscode on a different cluster and got this error:

```python
>>> import pyoscode
RuntimeError: module compiled against API version 0xe but this version of numpy is 0xd
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/lh561/Documents/Projects/OscodePrj/oscode/pyoscode/__init__.py", line 3, in <module>
    import _pyoscode
ImportError: numpy.core.multiarray failed to import
```

Updating from `numpy==1.19.5` to `numpy>=1.20` solved this issue for me.

### PR DESCRIPTION
**Pull request by lukashergt**
* Change `requirements.txt` to require `numpy>=1.20.0`.
* drop Python 2.7 support and support Python 3.8 and 3.9 instead 
  (note that most major packages like numpy and scipy have dropped Python 2.7 support now...)

fixes #8

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ def readme(short=False):
 
 setup(
     name="pyoscode",
-    version="1.0.2",
+    version="1.0.4",
```
Comment (lukashergt) [inline setup.py:23]: Is `setup.py` the only place that records the oscode version number? Or does that need updating somewhere else?

File: .travis.yml [hunk 2]
```diff
@@ -3,8 +3,9 @@ os:
     - linux
 osx_image:  xcode12.2
 python:
-    - "2.7"
     - "3.7"
+    - "3.8"
+    - "3.9"
 install: 
     - pip install -r requirements.txt
     - pip install pyoscode==1.0.2
```
Comment (lukashergt) [inline .travis.yml]: Wait, isn't `pip install pyoscode==1.0.2` problematic? That won't install the current version (the one that is supposed to be tested)... Shouldn't this be `pip install .`?

File: .travis.yml [hunk 3]
```diff
@@ -3,11 +3,14 @@ os:
     - linux
 osx_image:  xcode12.2
 python:
-    - "2.7"
     - "3.7"
+    - "3.8"
+    - "3.9"
```
Comment (lukashergt) [inline .travis.yml]: This entire section appears to be completely ignored by the Travis CI. See also #10.

Comment (fruzsinaagocs) [reply .travis.yml]: Indeed it was - on Mac OS, the `python: x.y` command was ignored, and the system default (2.7) was used instead. Moreover, `python --version` always returns 2.7. This issue was fixed in d3bb2ce.

File: .travis.yml (hunk 2 above)
Comment (fruzsinaagocs) [reply .travis.yml]: Yes, thank you for spotting that!

File: setup.py (hunk 1 above)
Comment (fruzsinaagocs) [reply setup.py:23]: The version number is recorded in `README.rst` and in `setup.py`.

**Review by fruzsinaagocs — APPROVED** [review]
Thank you for submitting this pull request! All changes are OK, but I further changed `.travis.yml` to properly test on Mac OS and Linux Xenial with Python 3.x.

File: setup.py (hunk 1 above)
Comment (lukashergt) [reply setup.py:23]: Wait, where in the README? Do you just mean the changelog?

Comment (fruzsinaagocs) [reply setup.py:23]: Yes, that's what I mean.

**Pull request merged**